src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -623,7 +623,7 @@ public CompilationInfo(Compilation compilation, bool hasSuccessfullyLoaded)
                                 var previousSubmissionCompilation =
                                     await solution.GetCompilationAsync(projectReference.ProjectId, cancellationToken).ConfigureAwait(false);
 
-                                compilation = compilation.WithScriptCompilationInfo(
+                                compilation = compilation!.WithScriptCompilationInfo(
                                     compilation.ScriptCompilationInfo.WithPreviousScriptCompilation(previousSubmissionCompilation));
```

Comment:
When is `compilation == null`?

Reply:
F# and TypeScript. I really wish we had a non-null sentinel for these cases.